android/app/src/main/kotlin/net/defined/mobile_nebula/NebulaVpnService.kt

```diff
@@ -209,11 +209,11 @@ class NebulaVpnService : VpnService() {
             }
         }
 
-        registerReceiver(receiver, IntentFilter(PowerManager.ACTION_DEVICE_IDLE_MODE_CHANGED))
```

Comment:
Is this one actually necessary?

> Exception for receivers that receive only system broadcasts

> If your app is registering a receiver only for [system broadcasts](https://developer.android.com/guide/components/broadcasts#system-broadcasts) through Context#registerReceiver methods, such as [Context#registerReceiver()](https://developer.android.com/reference/android/content/Context#registerReceiver(android.content.BroadcastReceiver,%20android.content.IntentFilter)), then it shouldn't specify a flag when registering the receiver.

Reply:
https://android.googlesource.com/platform/prebuilts/fullsdk/platforms/+/9698c0ff01d4b3e24cab840903c06a282db785b3/android-34/data/broadcast_actions.txt?pli=1#

`android.os.action.DEVICE_IDLE_MODE_CHANGED`

I think not?

Reply:
It sounds like the app doesn't built on API 34 without this flag and does with the addition of the flag though, right?

Reply:
@jasikpark That's not my reading of the first text I quoted, no.

Reply:
~~Sorry what I'm saying is it does sound like we don't need the flag from what the docs claim, but since all this does is change those flags, it isn't needed.~~

Realized that this was the unique receiver from the other two. disregard 🫡

Reply:
I'll test it out tomorrow.

Reply:
you were right, seems to work fine without it.